Yoast/wordpress-seo
# Moves permalink flushing for indexables to a helper

## Context


* By moving the `reset_permalink_indexables` into a helper, this allows up to reuse the method as part of the permalink safety project.

## Summary


This PR can be summarized in the following changelog entry:

* Moves the `get_permalink_for_indexable` into a permalink helper class for easier reuse.
* Moves the `reset_permalink_indexables` into the indexable helper class for easier reuse.

## Relevant technical choices:

* `get_permalink_for_indexable` was moved into its own helper to avoid cyclomatic dependencies that would otherwise occur after moving and refactoring methods.

## Test instructions

This PR can be tested by following these steps:

* Ensure you've run the indexation of post, terms etc. (i.e. SEO -> Tools -> Start processing and speed up your site now) if you haven't run it already.
* Verify that `permalink` and `permalink_hash` columns are filled when checking the `wp_yoast_indexable` table.
* Go into Settings -> Permalinks and alter the permalink structure. Hit save.
* Verify that `permalink` and `permalink_hash` columns are `null` when checking the `wp_yoast_indexable` table.

* Check out this PR and make sure you build everything.
* Re-run the above steps and verify that the results are the same as before checking out this PR.

## UI changes

## Documentation

## Quality assurance


Fixes [P3-159]
